packages/expo-dev-client/plugin/src/withGeneratedAndroidScheme.ts

```diff
@@ -24,5 +29,58 @@ export function setGeneratedAndroidScheme(
   if (!AndroidConfig.Scheme.hasScheme(scheme, androidManifest)) {
     androidManifest = AndroidConfig.Scheme.appendScheme(scheme, androidManifest);
   }
+
+  return androidManifest;
+}
+
+/**
+ * Remove the custom Expo dev client scheme from intent filters, which are set to `autoVerify=true`.
+ * The custom scheme `<data android:scheme="exp+<slug>"/>` seems to block verification for these intent filters.
+ * This plugin makes sure there is no scheme in the autoVerify intent filters, that starts with `exp+`.
+ 
+ * Iterate over all `autoVerify=true` intent filters, and pull out schemes starting with `exp+`.
+ *
+ * @param {AndroidManifest} androidManifest
+ */
+export function removeExpoSchemaFromVerifiedIntentFilters(androidManifest: AndroidManifest) {
+  // see: https://github.com/expo/expo-cli/blob/f1624c75b52cc1c4f99354ec4021494e0eff74aa/packages/config-plugins/src/android/Scheme.ts#L164-L179
+  for (const application of androidManifest.manifest.application || []) {
+    for (const activity of application.activity || []) {
+      if (activityHasSingleTaskLaunchMode(activity)) {
+        for (const intentFilter of activity['intent-filter'] || []) {
+          if (intentFilterHasAutoVerification(intentFilter) && intentFilter?.data) {
+            intentFilter.data = intentFilterRemoveSchemeFromData(intentFilter, (scheme: string) =>
+              scheme?.startsWith('exp+')
```

Comment:
would there be any downside to matching the exact scheme we just added, rather than any scheme that starts with `exp+`?

Reply:
I think in the issue there was some talk about moving this into config plugins as well, so there will likely be a follow up to this.

I can't think of any downside of matching exactly, I think I just copied cedrics original logic. I'll update it

Reply:
Testing this

Reply:
Another test